arviz/tests/base_tests/test_data.py

```diff
@@ -325,34 +335,23 @@ def test_addition(self):
         assert not fails
 
     @pytest.mark.parametrize("inplace", [True, False])
-    def test_extend_xr_method(self, inplace):
-        data = np.random.normal(size=(4, 500, 8))
-        dataset = from_dict(
-            posterior={"a": data[..., 0], "b": data},
-            prior={"a": data[..., 0], "b": data},
-            posterior_predictive={"a": data[..., 0], "b": data},
-        )
-        dataset_copy = dataset
+    def test_extend_xr_method(self, data_random, inplace):
+        idata = data_random
+        idata_copy = idata
```

Comment:
I think idata copy will not be a copy which means that the asserts below `assert_identical` are not asserting what we want to check, instead `assert_identical(idata_copy.posterior, idata.posterior)` would work, and therefore the `sum` is actually having no effect. 

In addition to using `.copy()` (I think idata already has the method), it is probably better to not use `sum()` but to use `sum(dim="draw")`, therefore in the situation I am talking about above, test would fail due to `draw` dim not being present

Reply:
Okay, I understand. But I don't think we have the `.copy` method for idata (yet).